text/0195-associated-items.md

```diff
@@ -1190,6 +1197,73 @@ trait Mappable
 While the above demonstrates the versatility of associated types and `where`
 clauses, it is probably too much of a hack to be viable for use in `libstd`.
 
+### Associated consts in generic code
+
+There are some restrictions on uses of associated consts in generic code. These
+might be loosened or removed in the future (see the related sub-sections in
+"Unresolved questions" below).
+
+ 1. Values of constant expressions in match patterns cannot depend on a type
+    parameter (by extension, neither can the types of such expressions). This
+    restriction is necessary for exhaustiveness and reachability to be checked
+    in generic code.
+
+    Note that the dependence of a value on a type parameter may be indirect:
+
+    ```rust
+    enum MyEnum {
+        Var1,
+        Var2,
+    }
+    trait HasVar {
+        const VAR: MyEnum;
+    }
+    fn do_something<T: HasVar>(x: MyEnum) {
+        const y: MyEnum = <T>::VAR;
```

Comment:
That's true for functions, perhaps, but I really don't see why this should be the case for constants. Constants can't introduce type parameters of their own, and lifetime parameters would be meaningless anyway since everything must be `'static`. So there's no actual nesting here; constants just sit in their containing scope like expressions, AFAICT.